### PR TITLE
Updating the imagemosaic docs

### PR DESCRIPTION
**Pull request by bmmpxf**

### CODE REVIEW

File: doc/en/user/source/data/raster/imagemosaic/configuration.rst [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+Granules
+--------
+
+Each individual image is commonly referred to as a **granule**. Individual granules do not need to share the same CRS or resolution, assuming appropriate configuration.
+
+There are some limitations:
+
+* All the granules must share the same ColorModel and SampleModel. This means that the granules must share the same pixel layout and photometric interpretation.
+
+* If overviews are used, the set of overviews must be the same for all the granules, otherwise the overviews will be ignored).
```
Comment (bradh) [inline doc/en/user/source/data/raster/imagemosaic/configuration.rst]: Stray bracket at the end.

File: doc/en/user/source/data/raster/imagemosaic/configuration.rst [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+     - Y
+     - The name of the attribute path in the shapefile index. Default is ``location``.
+   * - SuggestedSPI
+     - Y
+     - Suggested plugin for reading the image files.
+   * - Envelope2D
+     - Y
+     - Envelope for the mosaic formatted as ```LLX,LLY URX,URY`` (notice the space between the lower left and upper right coordinate pairs).
+   * - CheckAuxiliaryMetadata
+     - ???
```
Comment (bradh) [inline doc/en/user/source/data/raster/imagemosaic/configuration.rst]: We shouldn't put this in the docs.

File: doc/en/user/source/data/raster/imagemosaic/configuration.rst [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+:file:`datastore.properties`
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If the index is specified by a shapefile, it must be included in the root of the imagemosaic directory, just like the primary configuration file.
+
+Otherwise, the index connection parameters should be specified by a :file:`datastore.properties` file.
+
+.. note: A shapefile is created automatically if it does not exist or if there is no :file:`datastore.properties` file.
+
+.. Make sure a shapefile isn't created no matter what
```
Comment (bradh) [inline doc/en/user/source/data/raster/imagemosaic/configuration.rst]: This part isn't clear.

File: doc/en/user/source/data/raster/imagemosaic/configuration.rst [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+     - Description
+   * - SPI
+     - Y
+     - The DataStoreFactory used to connect to the index store:
+       
+       * PostGIS: ``org.geotools.data.postgis.PostgisNGDataStoreFactory`` 
+       * Oracle: ``org.geotools.data.oracle.OracleNGDataStoreFactory`` 
+       * H2: ``org.geotools.data.h2.H2DataStoreFactory``
+
+       :ref:`JNDI <tomcat_jndi>` can also be used with any of these stores. If JNDI is used, the DataStoreFactory name differ from the above.
```
Comment (bradh) [inline doc/en/user/source/data/raster/imagemosaic/configuration.rst]: differ -> differs

**Comment by simboss** [comment]
Opened a [new one](https://github.com/geoserver/geoserver/pull/1758) with my work in it.

**Comment by simboss** [comment]
Btw, thanks for the hard work!

**Pull request closed without merging**